zookeeper-server/src/main/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProvider.java

```diff
@@ -136,12 +136,12 @@ public boolean isAuthenticated() {
 
   @Override
   public boolean isValid(String id) {
-    try {
-      new X500Principal(id);
-      return true;
-    } catch (IllegalArgumentException e) {
-      return false;
-    }
+    // Id can be of multiple format since it can be either a domain or a client URI,
+    // so the check on Id format can be expensive.
+    // For users, the Id to be set is extracted by server therefore it must be of valid format.
+    // Only superusers can manually set ACL, who we should trust.
+    // Therefore it doesn't seem necessary to perform this check.
```

Comment:
This is good, could you just make sure we point to the function where we're checking if the given `id` is valid (the extraction logic)?

Also, it is either necessary or not - probably remove the wording that says "it seems/doesn't seem necessary..."